src/js/components/editor/Editor.vue

```diff
@@ -0,0 +1,284 @@
+             */
+            ssp: new SignStreamProcessor(),
+
+            /**
+             * @type {string} the hidden input value
+             */
+            input: '',
+
+            /**
+             * @type {Sign}  The sign instance currently in focus
+             */
+            focusedSign: null,
+
+            /**
+             * @type {Store} A Vuex store for the editor component tree
+             */
+            state: editorStore()
+        }
+    },
+    methods: {
+
+        /**
+         * Asynchronously retrieve the text from the server
+         * 
+         * @param {string} scrollVersionID The scroll ID
+         * @param {string} colID           The Column ID
+         */
+        getText(scrollVersionID, colID) {
+            this.$post('resources/cgi-bin/scrollery-cgi.pl', {
+                transaction: 'getSignStreamOfColumn',
+                SCROLL_VERSION: scrollVersionID,
+                colId: colID,
+                SESSION_ID: this.$store.getters.sessionID
+            })
+            .then(res => {
+                if (res.status === 200 && res.data) {
+                    return this.ssp.streamToTree(
+                        res.data.results, 
+                        'prev_sign_id',
+                        'sign_id',
+                        'next_sign_id'
+                    )
+                } else {
+                    throw new Error("Unable to retrieve text data")
+                }
+            })
+            .then(formattedNodes => {
+                this.currentText = formattedNodes
+            })
+            .catch(err => {
+                // TODO: handle err
+                console.error(err);
+            })
+        },
+
+        /**
+         * When the editor is clicked on, we set focus on the hidden editor input
+         * 
+         * @param {Sign} sign The sign clicked on
+         */
+        onClick(sign) {
+            this.focusedSign = sign
+            document.getElementById('editor-input').focus()
+        },
+
+        /**
+         * @param {FocusEvent} event  The keyboard event received from the input
+         */
+         onInputBlur(event) {
+            this.focusedSign = null;
+        },
+
+        /**
+         * @param {KeyboardEvent} event  The keyboard event received from the input
+         */
+        onKeyup(event) {
+            switch(event.code) {
+
+                // next sign
+                case KEY_CODES.ARROWS.LEFT:
+                    if (this.focusedSign && this.focusedSign.hasNext()) {
+                        this.focusedSign = this.focusedSign.next();
+                    }
+                    break;
+
+                // previous sign
+                case KEY_CODES.ARROWS.RIGHT:
+                    if (this.focusedSign && this.focusedSign.hasPrevious()) {
+                        this.focusedSign = this.focusedSign.previous();
+                    }
+                    break;
+                    break;
+            }
+
```

Comment:
We add "up" and "down" also perhaps at some point.  I think you just count how many characters come before the cursor in its current line then move the cursor up or down a line, then to the left that same number of characters (or to the end of the line if it doesn't have that many characters).  I guess you also need to do nothing when someone hits up on the first line or down on the last.  (I imagine you probably had something like this already figured out.)

Reply:
Yes, that's certainly on this list. My thinking is more or less aligned with your idea on how to handle.